### PR TITLE
Add Slovakia holidays

### PR DESCRIPTION
**Pull request by patressz**
I did have a problem with encoding, so i add test snapshot with special characters.

### CODE REVIEW

File: tests/.pest/snapshots/Countries/SlovakiaTest/it_can_calculate_slovak_holidays.snap [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        "name": "De\u0148 v\u00ed\u0165azstva nad fa\u0161izmom",
+        "date": "2024-05-08"
+    },
+    {
+        "name": "Sviatok sv\u00e4t\u00e9ho Cyrila a Metoda",
+        "date": "2024-07-05"
+    },
+    {
+        "name": "V\u00fdro\u010die V\u00fdro\u010die Slovensk\u00e9ho n\u00e1rodn\u00e9ho povstania",
```
Comment (matusk) [inline tests/.pest/snapshots/Countries/SlovakiaTest/it_can_calculate_slovak_holidays.snap:31]:
```suggestion
        "name": "V\u00fdro\u010die Slovensk\u00e9ho n\u00e1rodn\u00e9ho povstania",
```

File: src/Countries/Slovakia.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return array_merge([
+            'Deň vzniku Slovenskej republiky' => '01-01',
+            'Zjavenie Pána (Traja králi)' => '01-06',
+            'Sviatok práce' => '05-01',
+            'Deň víťazstva nad fašizmom' => '05-08',
+            'Sviatok svätého Cyrila a Metoda' => '07-05',
+            'Výročie Výročie Slovenského národného povstania' => '08-29',
+            'Deň Ústavy Slovenskej republiky' => '09-01',
+            'Sedembolestná Panna Mária' => '09-15',
+            'Deň vzniku samostatného česko-slovenského štátu' => '10-28',
```
Comment (matusk) [inline src/Countries/Slovakia.php:25]:
```suggestion
```
this is not public holiday

File: tests/.pest/snapshots/Countries/SlovakiaTest/it_can_calculate_slovak_holidays.snap [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        "date": "2024-09-01"
+    },
+    {
+        "name": "Sedembolestn\u00e1 Panna M\u00e1ria",
+        "date": "2024-09-15"
+    },
+    {
+        "name": "De\u0148 vzniku samostatn\u00e9ho \u010desko-slovensk\u00e9ho \u0161t\u00e1tu",
+        "date": "2024-10-28"
+    },
```
Comment (matusk) [inline tests/.pest/snapshots/Countries/SlovakiaTest/it_can_calculate_slovak_holidays.snap:45]:
```suggestion
```
this is not public holiday

**Review by matusk — CHANGES_REQUESTED** [review]

File: src/Countries/Slovakia.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            'Štedrý deň' => '12-24',
+            'Prvý sviatok vianočný' => '12-25',
+            'Druhý sviatok vianočný' => '12-26',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
```
Comment (Nielsvanpach) [inline src/Countries/Slovakia.php:36]: Could you use `$this->easter($year)` instead?

**Review by Nielsvanpach — COMMENTED** [review]
Thanks! PR looks good. I'll wait on some approvals by others before merging.

File: src/Countries/Slovakia.php (hunk 4 above)
Comment (patressz) [reply src/Countries/Slovakia.php]: Of course, fixed.

**Review by RobiNN1 — APPROVED** [review]

**Comment by Nielsvanpach** [comment]
Thanks!

**Pull request merged**